src/Windows/Avalonia.Win32/DataTransferToOleDataObjectWrapper.cs

```diff
@@ -0,0 +1,202 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Runtime.InteropServices.ComTypes;
+using Avalonia.Input;
+using Avalonia.MicroCom;
+using Avalonia.Platform.Storage;
+using static Avalonia.Win32.Interop.UnmanagedMethods;
+using FORMATETC = Avalonia.Win32.Interop.UnmanagedMethods.FORMATETC;
+using STGMEDIUM = Avalonia.Win32.Interop.UnmanagedMethods.STGMEDIUM;
+
+namespace Avalonia.Win32;
+
+/// <summary>
+/// Wraps an Avalonia <see cref="IDataTransfer"/> into a Win32 <see cref="Win32Com.IDataObject"/>.
+/// </summary>
+/// <param name="dataTransfer">The wrapped data transfer instance.</param>
+internal class DataTransferToOleDataObjectWrapper(IDataTransfer dataTransfer)
```

Comment:
Do you think we can support [IDataObjectAsyncCapability](https://learn.microsoft.com/en-us/windows/win32/api/shldisp/nn-shldisp-idataobjectasynccapability) now?

Reply:
See https://github.com/dotnet/winforms/issues/13422 too

Reply:
Yes we should be able to, but I'd rather have that as another PR.